framework/src/auxkernels/ProjectionAux.C

```diff
@@ -44,11 +49,17 @@ ProjectionAux::ProjectionAux(const InputParameters & parameters)
   : AuxKernel(parameters),
     _v(coupledValue("v")),
     _source_variable(*getFieldVar("v", 0)),
-    _source_sys(_c_fe_problem.getSystem(coupledName("v")))
+    _source_sys(_c_fe_problem.getSystem(coupledName("v"))),
+    _use_block_restriction_for_source(getParam<bool>("use_block_restriction_for_source"))
 {
   // Output some messages to user
   if (_source_variable.order() > _var.order())
     mooseInfo("Projection lowers order, please expect a loss of accuracy");
+  // Check the dimension of the block restriction
+  for (const auto & sub_id : blockIDs())
+    if (_mesh.isLowerD(sub_id))
+      paramError("block",
+                 "ProjectionAux's block restriction must not include lower dimensional blocks");
```

Comment:
so we can't do projection of lower-dimensional variables?

Reply:
seems like we should figure out the root cause, e.g. the poor ghosting, in #28632

Reply:
The OP wasn't trying to, they just happened to have lower and higher dimensional elements

Reply:
but this will now prevent people from doing projection of lower-dimensional variables instead of enabling doing it without segfaults